ReportApp.Web/Controllers/ManagementController.cs

```diff
@@ -115,5 +130,78 @@ public JsonResult UnitListByDepartmentId(int id)
         {
             return Json(new SelectList(_unitRepository.GetUnitListByDepartmentId(id).ToArray(), "UnitId", "UnitName"), JsonRequestBehavior.AllowGet);
         }
+
+        public async Task<ActionResult> AssignRole(string id)
+        {
+            if (id == null)
+            {
+                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
+            }
+            var profile = _staffRepository.GetProfileById(id);
+            if (profile == null)
+            {
+                return HttpNotFound();
+            }
+            var userRoles = await UserManager.GetRolesAsync(id);
+            
+            return View(new AssignRoleModel
+            {
+                Id = profile.Staff.Id,
+                Name = profile.FullName,
+                Department = profile.Unit.Department.DepartmentName,
+                Unit =  profile.Unit.UnitName,
+                RolesList = RoleManager.Roles.ToList().Select(x => new SelectListItem()
+                {
+                    Selected = userRoles.Contains(x.Name),
+                    Text = x.Name,
+                    Value = x.Name
+                })
+            });
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<ActionResult> AssignRole([Bind(Include = "Id")] AssignRoleModel roleModel, params string[] selectedRoles)
+        {
+            if (ModelState.IsValid)
+            {
+                var profile = _staffRepository.GetProfileById(roleModel.Id);
+                if (profile == null)
```

Comment:
if profile == null can be DRY-ed out. Consider having an utility function for all these...